{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "RNA-seq data is in log2 of TMM.<br>\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2864565/ <br>\n",
    "https://www.biostars.org/p/317417/ <br>\n",
    "Proteomics is in log2 intensity, where intensity is AUC (range from 10^6-10^9)<br>\n",
    "https://en.wikipedia.org/wiki/Quantitative_proteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Furthermore not sure whether or not to do thresholding on the values. In the parsed tables there is no thresholding however."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Table containing all of the rna-seq data for both cell line and all ages, includes ~40k genes. Need to remove zero rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene.names</th>\n",
       "      <th>Gene.description</th>\n",
       "      <th>AvgChrs</th>\n",
       "      <th>AT2_04M_F0</th>\n",
       "      <th>AT2_18M_F0</th>\n",
       "      <th>AT2_04M_F10</th>\n",
       "      <th>AT2_18M_F10</th>\n",
       "      <th>AM_04M_F0</th>\n",
       "      <th>AM_18M_F0</th>\n",
       "      <th>AM_04M_F10</th>\n",
       "      <th>AM_18M_F10</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Row.names</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>Gnai3</td>\n",
       "      <td>guanine nucleotide binding protein (G protein)...</td>\n",
       "      <td>2.675624</td>\n",
       "      <td>7.151432</td>\n",
       "      <td>7.134999</td>\n",
       "      <td>7.301802</td>\n",
       "      <td>7.126699</td>\n",
       "      <td>7.808586</td>\n",
       "      <td>7.735874</td>\n",
       "      <td>7.599455</td>\n",
       "      <td>7.866372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000003</td>\n",
       "      <td>Pbsn</td>\n",
       "      <td>probasin</td>\n",
       "      <td>19.455155</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>Cdc45</td>\n",
       "      <td>cell division cycle 45</td>\n",
       "      <td>15.191392</td>\n",
       "      <td>1.347537</td>\n",
       "      <td>0.520289</td>\n",
       "      <td>1.752669</td>\n",
       "      <td>1.091367</td>\n",
       "      <td>2.649199</td>\n",
       "      <td>1.417394</td>\n",
       "      <td>3.991956</td>\n",
       "      <td>3.696447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000031</td>\n",
       "      <td>H19</td>\n",
       "      <td>H19, imprinted maternally expressed transcript</td>\n",
       "      <td>6.980304</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000037</td>\n",
       "      <td>Scml2</td>\n",
       "      <td>Scm polycomb group protein like 2</td>\n",
       "      <td>19.942446</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene.names  \\\n",
       "Row.names                       \n",
       "ENSMUSG00000000001      Gnai3   \n",
       "ENSMUSG00000000003       Pbsn   \n",
       "ENSMUSG00000000028      Cdc45   \n",
       "ENSMUSG00000000031        H19   \n",
       "ENSMUSG00000000037      Scml2   \n",
       "\n",
       "                                                     Gene.description  \\\n",
       "Row.names                                                               \n",
       "ENSMUSG00000000001  guanine nucleotide binding protein (G protein)...   \n",
       "ENSMUSG00000000003                                           probasin   \n",
       "ENSMUSG00000000028                             cell division cycle 45   \n",
       "ENSMUSG00000000031     H19, imprinted maternally expressed transcript   \n",
       "ENSMUSG00000000037                  Scm polycomb group protein like 2   \n",
       "\n",
       "                      AvgChrs  AT2_04M_F0  AT2_18M_F0  AT2_04M_F10  \\\n",
       "Row.names                                                            \n",
       "ENSMUSG00000000001   2.675624    7.151432    7.134999     7.301802   \n",
       "ENSMUSG00000000003  19.455155    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000028  15.191392    1.347537    0.520289     1.752669   \n",
       "ENSMUSG00000000031   6.980304    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000037  19.942446    0.000000    0.000000     0.000000   \n",
       "\n",
       "                    AT2_18M_F10  AM_04M_F0  AM_18M_F0  AM_04M_F10  AM_18M_F10  \n",
       "Row.names                                                                      \n",
       "ENSMUSG00000000001     7.126699   7.808586   7.735874    7.599455    7.866372  \n",
       "ENSMUSG00000000003     0.000000   0.000000   0.000000    0.000000    0.000000  \n",
       "ENSMUSG00000000028     1.091367   2.649199   1.417394    3.991956    3.696447  \n",
       "ENSMUSG00000000031     0.000000   0.000000   0.000000    0.000000    0.000000  \n",
       "ENSMUSG00000000037     0.000000   0.000000   0.000000    0.000000    0.000000  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_seq = pd.read_csv('../../Data/RawData/PPG_RNAseq_covariance_input_0419.tsv',sep='\\t',index_col=0)\n",
    "rna_seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Proteomics Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               AM_18M_F0_P  AM_04M_F0_P  AM_18M_F10_P  AM_04M_F10_P\n",
      "Gene.names                                                         \n",
      "1600014C10Rik    24.555314    24.870091     25.209116     25.126227\n",
      "1700037H04Rik    24.101137    24.399008     23.703847     23.697930\n",
      "1810009A15Rik    24.519323    24.825104     24.583232     24.895794\n",
      "1810030O07Rik    22.569708    22.366676     22.634868      0.000000\n",
      "1810037I17Rik     0.000000    24.252658      0.000000      0.000000\n",
      "(4017, 4)\n",
      "               AT2_18M_F0_P  AT2_04M_F0_P  AT2_18M_F10_P  AT2_04M_F10_P\n",
      "Gene.names                                                             \n",
      "0610037L13Rik     23.812841     23.464822      23.808293      23.734857\n",
      "1810009A15Rik     24.068330     23.779004      23.182809      23.169902\n",
      "2310011J03Rik     23.167732     23.107635      23.487317      23.347608\n",
      "2610301B20Rik     23.308364      0.000000       0.000000       0.000000\n",
      "2810006K23Rik     21.731796      0.000000       0.000000       0.000000\n",
      "(3866, 4)\n"
     ]
    }
   ],
   "source": [
    "proteomics_AM = pd.read_csv('../../Data/RawData/PPG_AM_proteomics_covariance_input_0419.tsv',sep='\\t',index_col=0)\n",
    "proteomics_AM.columns = ['AvgChrs','AM_18M_F0_P','AM_04M_F0_P','AM_18M_F10_P','AM_04M_F10_P']\n",
    "proteomics_AM.drop(columns='AvgChrs',inplace=True)\n",
    "print(proteomics_AM.head())\n",
    "print(proteomics_AM.shape)\n",
    "\n",
    "proteomics_AT = pd.read_csv('../../Data/RawData/PPG_AT2_proteomics_covariance_input_0419.tsv',sep='\\t',index_col=0)\n",
    "proteomics_AT.columns = ['AvgChrs','AT2_18M_F0_P','AT2_04M_F0_P','AT2_18M_F10_P','AT2_04M_F10_P']\n",
    "proteomics_AT.drop(columns='AvgChrs',inplace=True)\n",
    "print(proteomics_AT.head())\n",
    "print(proteomics_AT.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference chromosome lengths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "      <th>a</th>\n",
       "      <th>b</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chromosome</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>195471971</td>\n",
       "      <td>CM000994.2</td>\n",
       "      <td>NC_000067.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>182113224</td>\n",
       "      <td>CM000995.2</td>\n",
       "      <td>NC_000068.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>160039680</td>\n",
       "      <td>CM000996.2</td>\n",
       "      <td>NC_000069.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>156508116</td>\n",
       "      <td>CM000997.2</td>\n",
       "      <td>NC_000070.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>151834684</td>\n",
       "      <td>CM000998.2</td>\n",
       "      <td>NC_000071.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>149736546</td>\n",
       "      <td>CM000999.2</td>\n",
       "      <td>NC_000072.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>145441459</td>\n",
       "      <td>CM001000.2</td>\n",
       "      <td>NC_000073.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>129401213</td>\n",
       "      <td>CM001001.2</td>\n",
       "      <td>NC_000074.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>124595110</td>\n",
       "      <td>CM001002.2</td>\n",
       "      <td>NC_000075.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>130694993</td>\n",
       "      <td>CM001003.2</td>\n",
       "      <td>NC_000076.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>122082543</td>\n",
       "      <td>CM001004.2</td>\n",
       "      <td>NC_000077.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>120129022</td>\n",
       "      <td>CM001005.2</td>\n",
       "      <td>NC_000078.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>120421639</td>\n",
       "      <td>CM001006.2</td>\n",
       "      <td>NC_000079.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>124902244</td>\n",
       "      <td>CM001007.2</td>\n",
       "      <td>NC_000080.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>104043685</td>\n",
       "      <td>CM001008.2</td>\n",
       "      <td>NC_000081.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>98207768</td>\n",
       "      <td>CM001009.2</td>\n",
       "      <td>NC_000082.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>94987271</td>\n",
       "      <td>CM001010.2</td>\n",
       "      <td>NC_000083.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>90702639</td>\n",
       "      <td>CM001011.2</td>\n",
       "      <td>NC_000084.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>61431566</td>\n",
       "      <td>CM001012.2</td>\n",
       "      <td>NC_000085.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>X</td>\n",
       "      <td>171031299</td>\n",
       "      <td>CM001013.2</td>\n",
       "      <td>NC_000086.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Y</td>\n",
       "      <td>91744698</td>\n",
       "      <td>CM001014.2</td>\n",
       "      <td>NC_000087.7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               length           a            b\n",
       "Chromosome                                    \n",
       "1           195471971  CM000994.2  NC_000067.6\n",
       "2           182113224  CM000995.2  NC_000068.7\n",
       "3           160039680  CM000996.2  NC_000069.6\n",
       "4           156508116  CM000997.2  NC_000070.6\n",
       "5           151834684  CM000998.2  NC_000071.6\n",
       "6           149736546  CM000999.2  NC_000072.6\n",
       "7           145441459  CM001000.2  NC_000073.6\n",
       "8           129401213  CM001001.2  NC_000074.6\n",
       "9           124595110  CM001002.2  NC_000075.6\n",
       "10          130694993  CM001003.2  NC_000076.6\n",
       "11          122082543  CM001004.2  NC_000077.6\n",
       "12          120129022  CM001005.2  NC_000078.6\n",
       "13          120421639  CM001006.2  NC_000079.6\n",
       "14          124902244  CM001007.2  NC_000080.6\n",
       "15          104043685  CM001008.2  NC_000081.6\n",
       "16           98207768  CM001009.2  NC_000082.6\n",
       "17           94987271  CM001010.2  NC_000083.6\n",
       "18           90702639  CM001011.2  NC_000084.6\n",
       "19           61431566  CM001012.2  NC_000085.6\n",
       "X           171031299  CM001013.2  NC_000086.7\n",
       "Y            91744698  CM001014.2  NC_000087.7"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrom_info = pd.read_csv('../../Data/RawData/ChromosomeLength_MmG38p6.tsv',sep='\\t',index_col=0)\n",
    "chrom_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference start/end locations as well as the chromosome number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000064372</td>\n",
       "      <td>15356</td>\n",
       "      <td>15422</td>\n",
       "      <td>MT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000064371</td>\n",
       "      <td>15289</td>\n",
       "      <td>15355</td>\n",
       "      <td>MT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000064370</td>\n",
       "      <td>14145</td>\n",
       "      <td>15288</td>\n",
       "      <td>MT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000064369</td>\n",
       "      <td>14071</td>\n",
       "      <td>14139</td>\n",
       "      <td>MT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000064368</td>\n",
       "      <td>13552</td>\n",
       "      <td>14070</td>\n",
       "      <td>MT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Gene start (bp)  Gene end (bp) Chromosome/scaffold name\n",
       "Gene stable ID                                                             \n",
       "ENSMUSG00000064372            15356          15422                       MT\n",
       "ENSMUSG00000064371            15289          15355                       MT\n",
       "ENSMUSG00000064370            14145          15288                       MT\n",
       "ENSMUSG00000064369            14071          14139                       MT\n",
       "ENSMUSG00000064368            13552          14070                       MT"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sal_EnsembleIds_StartEnd = pd.read_csv('../../Data/RawData/mart_export_mm_gr38_82718.txt',sep='\\t',index_col=0)\n",
    "sal_EnsembleIds_StartEnd.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine rna-seq with gene start/end and lengths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene.names</th>\n",
       "      <th>Gene.description</th>\n",
       "      <th>AvgChrs</th>\n",
       "      <th>AT2_04M_F0</th>\n",
       "      <th>AT2_18M_F0</th>\n",
       "      <th>AT2_04M_F10</th>\n",
       "      <th>AT2_18M_F10</th>\n",
       "      <th>AM_04M_F0</th>\n",
       "      <th>AM_18M_F0</th>\n",
       "      <th>AM_04M_F10</th>\n",
       "      <th>AM_18M_F10</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene Length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Row.names</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>Gnai3</td>\n",
       "      <td>guanine nucleotide binding protein (G protein)...</td>\n",
       "      <td>2.675624</td>\n",
       "      <td>7.151432</td>\n",
       "      <td>7.134999</td>\n",
       "      <td>7.301802</td>\n",
       "      <td>7.126699</td>\n",
       "      <td>7.808586</td>\n",
       "      <td>7.735874</td>\n",
       "      <td>7.599455</td>\n",
       "      <td>7.866372</td>\n",
       "      <td>108107280</td>\n",
       "      <td>108146146</td>\n",
       "      <td>3</td>\n",
       "      <td>38866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000003</td>\n",
       "      <td>Pbsn</td>\n",
       "      <td>probasin</td>\n",
       "      <td>19.455155</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>77837901</td>\n",
       "      <td>77853623</td>\n",
       "      <td>X</td>\n",
       "      <td>15722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>Cdc45</td>\n",
       "      <td>cell division cycle 45</td>\n",
       "      <td>15.191392</td>\n",
       "      <td>1.347537</td>\n",
       "      <td>0.520289</td>\n",
       "      <td>1.752669</td>\n",
       "      <td>1.091367</td>\n",
       "      <td>2.649199</td>\n",
       "      <td>1.417394</td>\n",
       "      <td>3.991956</td>\n",
       "      <td>3.696447</td>\n",
       "      <td>18780447</td>\n",
       "      <td>18811987</td>\n",
       "      <td>16</td>\n",
       "      <td>31540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000031</td>\n",
       "      <td>H19</td>\n",
       "      <td>H19, imprinted maternally expressed transcript</td>\n",
       "      <td>6.980304</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>142575529</td>\n",
       "      <td>142578143</td>\n",
       "      <td>7</td>\n",
       "      <td>2614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000037</td>\n",
       "      <td>Scml2</td>\n",
       "      <td>Scm polycomb group protein like 2</td>\n",
       "      <td>19.942446</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>161117193</td>\n",
       "      <td>161258213</td>\n",
       "      <td>X</td>\n",
       "      <td>141020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene.names  \\\n",
       "Row.names                       \n",
       "ENSMUSG00000000001      Gnai3   \n",
       "ENSMUSG00000000003       Pbsn   \n",
       "ENSMUSG00000000028      Cdc45   \n",
       "ENSMUSG00000000031        H19   \n",
       "ENSMUSG00000000037      Scml2   \n",
       "\n",
       "                                                     Gene.description  \\\n",
       "Row.names                                                               \n",
       "ENSMUSG00000000001  guanine nucleotide binding protein (G protein)...   \n",
       "ENSMUSG00000000003                                           probasin   \n",
       "ENSMUSG00000000028                             cell division cycle 45   \n",
       "ENSMUSG00000000031     H19, imprinted maternally expressed transcript   \n",
       "ENSMUSG00000000037                  Scm polycomb group protein like 2   \n",
       "\n",
       "                      AvgChrs  AT2_04M_F0  AT2_18M_F0  AT2_04M_F10  \\\n",
       "Row.names                                                            \n",
       "ENSMUSG00000000001   2.675624    7.151432    7.134999     7.301802   \n",
       "ENSMUSG00000000003  19.455155    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000028  15.191392    1.347537    0.520289     1.752669   \n",
       "ENSMUSG00000000031   6.980304    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000037  19.942446    0.000000    0.000000     0.000000   \n",
       "\n",
       "                    AT2_18M_F10  AM_04M_F0  AM_18M_F0  AM_04M_F10  AM_18M_F10  \\\n",
       "Row.names                                                                       \n",
       "ENSMUSG00000000001     7.126699   7.808586   7.735874    7.599455    7.866372   \n",
       "ENSMUSG00000000003     0.000000   0.000000   0.000000    0.000000    0.000000   \n",
       "ENSMUSG00000000028     1.091367   2.649199   1.417394    3.991956    3.696447   \n",
       "ENSMUSG00000000031     0.000000   0.000000   0.000000    0.000000    0.000000   \n",
       "ENSMUSG00000000037     0.000000   0.000000   0.000000    0.000000    0.000000   \n",
       "\n",
       "                    Gene start (bp)  Gene end (bp) Chromosome/scaffold name  \\\n",
       "Row.names                                                                     \n",
       "ENSMUSG00000000001        108107280      108146146                        3   \n",
       "ENSMUSG00000000003         77837901       77853623                        X   \n",
       "ENSMUSG00000000028         18780447       18811987                       16   \n",
       "ENSMUSG00000000031        142575529      142578143                        7   \n",
       "ENSMUSG00000000037        161117193      161258213                        X   \n",
       "\n",
       "                    Gene Length  \n",
       "Row.names                        \n",
       "ENSMUSG00000000001        38866  \n",
       "ENSMUSG00000000003        15722  \n",
       "ENSMUSG00000000028        31540  \n",
       "ENSMUSG00000000031         2614  \n",
       "ENSMUSG00000000037       141020  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assert all([gene in sal_EnsembleIds_StartEnd.index for gene in rna_seq.index]) #ensures that all of the gene indicie match\n",
    "rna_seq = rna_seq.join(sal_EnsembleIds_StartEnd)\n",
    "\n",
    "#Add gene lenth\n",
    "gene_length = pd.DataFrame(rna_seq['Gene end (bp)'] - rna_seq['Gene start (bp)'],columns=['Gene Length'])\n",
    "rna_seq = rna_seq.join(gene_length)\n",
    "rna_seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine rna-seq with proteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene.names</th>\n",
       "      <th>Gene.description</th>\n",
       "      <th>AvgChrs</th>\n",
       "      <th>AT2_04M_F0</th>\n",
       "      <th>AT2_18M_F0</th>\n",
       "      <th>AT2_04M_F10</th>\n",
       "      <th>AT2_18M_F10</th>\n",
       "      <th>AM_04M_F0</th>\n",
       "      <th>AM_18M_F0</th>\n",
       "      <th>AM_04M_F10</th>\n",
       "      <th>...</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>AM_18M_F0_P</th>\n",
       "      <th>AM_04M_F0_P</th>\n",
       "      <th>AM_18M_F10_P</th>\n",
       "      <th>AM_04M_F10_P</th>\n",
       "      <th>AT2_18M_F0_P</th>\n",
       "      <th>AT2_04M_F0_P</th>\n",
       "      <th>AT2_18M_F10_P</th>\n",
       "      <th>AT2_04M_F10_P</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Row.names</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>Gnai3</td>\n",
       "      <td>guanine nucleotide binding protein (G protein)...</td>\n",
       "      <td>2.675624</td>\n",
       "      <td>7.151432</td>\n",
       "      <td>7.134999</td>\n",
       "      <td>7.301802</td>\n",
       "      <td>7.126699</td>\n",
       "      <td>7.808586</td>\n",
       "      <td>7.735874</td>\n",
       "      <td>7.599455</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>38866</td>\n",
       "      <td>25.398656</td>\n",
       "      <td>25.29266</td>\n",
       "      <td>25.440647</td>\n",
       "      <td>25.149847</td>\n",
       "      <td>25.344119</td>\n",
       "      <td>25.38975</td>\n",
       "      <td>25.402401</td>\n",
       "      <td>25.485167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000003</td>\n",
       "      <td>Pbsn</td>\n",
       "      <td>probasin</td>\n",
       "      <td>19.455155</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>X</td>\n",
       "      <td>15722</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>Cdc45</td>\n",
       "      <td>cell division cycle 45</td>\n",
       "      <td>15.191392</td>\n",
       "      <td>1.347537</td>\n",
       "      <td>0.520289</td>\n",
       "      <td>1.752669</td>\n",
       "      <td>1.091367</td>\n",
       "      <td>2.649199</td>\n",
       "      <td>1.417394</td>\n",
       "      <td>3.991956</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>31540</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000031</td>\n",
       "      <td>H19</td>\n",
       "      <td>H19, imprinted maternally expressed transcript</td>\n",
       "      <td>6.980304</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>2614</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000037</td>\n",
       "      <td>Scml2</td>\n",
       "      <td>Scm polycomb group protein like 2</td>\n",
       "      <td>19.942446</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>X</td>\n",
       "      <td>141020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene.names  \\\n",
       "Row.names                       \n",
       "ENSMUSG00000000001      Gnai3   \n",
       "ENSMUSG00000000003       Pbsn   \n",
       "ENSMUSG00000000028      Cdc45   \n",
       "ENSMUSG00000000031        H19   \n",
       "ENSMUSG00000000037      Scml2   \n",
       "\n",
       "                                                     Gene.description  \\\n",
       "Row.names                                                               \n",
       "ENSMUSG00000000001  guanine nucleotide binding protein (G protein)...   \n",
       "ENSMUSG00000000003                                           probasin   \n",
       "ENSMUSG00000000028                             cell division cycle 45   \n",
       "ENSMUSG00000000031     H19, imprinted maternally expressed transcript   \n",
       "ENSMUSG00000000037                  Scm polycomb group protein like 2   \n",
       "\n",
       "                      AvgChrs  AT2_04M_F0  AT2_18M_F0  AT2_04M_F10  \\\n",
       "Row.names                                                            \n",
       "ENSMUSG00000000001   2.675624    7.151432    7.134999     7.301802   \n",
       "ENSMUSG00000000003  19.455155    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000028  15.191392    1.347537    0.520289     1.752669   \n",
       "ENSMUSG00000000031   6.980304    0.000000    0.000000     0.000000   \n",
       "ENSMUSG00000000037  19.942446    0.000000    0.000000     0.000000   \n",
       "\n",
       "                    AT2_18M_F10  AM_04M_F0  AM_18M_F0  AM_04M_F10  ...  \\\n",
       "Row.names                                                          ...   \n",
       "ENSMUSG00000000001     7.126699   7.808586   7.735874    7.599455  ...   \n",
       "ENSMUSG00000000003     0.000000   0.000000   0.000000    0.000000  ...   \n",
       "ENSMUSG00000000028     1.091367   2.649199   1.417394    3.991956  ...   \n",
       "ENSMUSG00000000031     0.000000   0.000000   0.000000    0.000000  ...   \n",
       "ENSMUSG00000000037     0.000000   0.000000   0.000000    0.000000  ...   \n",
       "\n",
       "                    Chromosome/scaffold name  Gene Length  AM_18M_F0_P  \\\n",
       "Row.names                                                                \n",
       "ENSMUSG00000000001                         3        38866    25.398656   \n",
       "ENSMUSG00000000003                         X        15722     0.000000   \n",
       "ENSMUSG00000000028                        16        31540     0.000000   \n",
       "ENSMUSG00000000031                         7         2614     0.000000   \n",
       "ENSMUSG00000000037                         X       141020     0.000000   \n",
       "\n",
       "                   AM_04M_F0_P  AM_18M_F10_P  AM_04M_F10_P  AT2_18M_F0_P  \\\n",
       "Row.names                                                                  \n",
       "ENSMUSG00000000001    25.29266     25.440647     25.149847     25.344119   \n",
       "ENSMUSG00000000003     0.00000      0.000000      0.000000      0.000000   \n",
       "ENSMUSG00000000028     0.00000      0.000000      0.000000      0.000000   \n",
       "ENSMUSG00000000031     0.00000      0.000000      0.000000      0.000000   \n",
       "ENSMUSG00000000037     0.00000      0.000000      0.000000      0.000000   \n",
       "\n",
       "                    AT2_04M_F0_P  AT2_18M_F10_P  AT2_04M_F10_P  \n",
       "Row.names                                                       \n",
       "ENSMUSG00000000001      25.38975      25.402401      25.485167  \n",
       "ENSMUSG00000000003       0.00000       0.000000       0.000000  \n",
       "ENSMUSG00000000028       0.00000       0.000000       0.000000  \n",
       "ENSMUSG00000000031       0.00000       0.000000       0.000000  \n",
       "ENSMUSG00000000037       0.00000       0.000000       0.000000  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_seq = rna_seq.join(proteomics_AM,on='Gene.names')\n",
    "rna_seq = rna_seq.join(proteomics_AT,on='Gene.names')\n",
    "rna_seq.fillna(0,inplace=True)\n",
    "rna_seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Recalculate avg chrm position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "chrom_lengths = dict(chrom_info['length'])\n",
    "chrom_nums = {val: num for num,val in enumerate(chrom_info.index)}\n",
    "\n",
    "avg_position = ((rna_seq['Gene end (bp)'] + rna_seq['Gene start (bp)'])/2/ rna_seq['Chromosome/scaffold name'].map(chrom_lengths))\n",
    "NormPosition = pd.DataFrame(avg_position,columns=['NormPosition'])\n",
    "chrom_pos = pd.DataFrame(rna_seq['Chromosome/scaffold name'].map(chrom_nums) + avg_position,columns=['AvgChrs2'])\n",
    "\n",
    "rna_seq = rna_seq.join(NormPosition)\n",
    "rna_seq = rna_seq.join(chrom_pos)\n",
    "\n",
    "if max(rna_seq['AvgChrs'] - rna_seq['AvgChrs2']) < 10e-6:\n",
    "    rna_seq.drop(columns='AvgChrs2',inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove genes with no values for both rna-seq or proteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_lines_rna = ['AT2_04M_F0','AT2_18M_F0','AT2_04M_F10','AT2_18M_F10','AM_04M_F0','AM_18M_F0','AM_04M_F10','AM_18M_F10']\n",
    "cell_lines_prot = ['AT2_04M_F0_P','AT2_18M_F0_P','AT2_04M_F10_P','AT2_18M_F10_P','AM_04M_F0_P','AM_18M_F0_P','AM_04M_F10_P','AM_18M_F10_P']\n",
    "cell_lines = cell_lines_rna+cell_lines_prot\n",
    "\n",
    "#Removes rows where all rna-seq values and all protein values are 0\n",
    "rna_seq = rna_seq[rna_seq[cell_lines].sum(axis=1)!=0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Missing protein values with known rna-seq values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_prot = rna_seq[rna_seq[cell_lines_prot].sum(axis=1)==0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Missing rna-seq with known protein values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_rna = rna_seq[rna_seq[cell_lines_rna].sum(axis=1)==0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove genes with missing rna-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "GP_data = rna_seq[rna_seq[cell_lines_rna].sum(axis=1)!=0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Saving tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_prot.to_csv(path_or_buf='../../Data/ProcessedData/missing_prot.tsv',sep='\\t')\n",
    "missing_rna.to_csv(path_or_buf='../../Data/ProcessedData/missing_rna.tsv',sep='\\t')\n",
    "GP_data.to_csv(path_or_buf='../../Data/ProcessedData/GP_data.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene.names</th>\n",
       "      <th>Gene.description</th>\n",
       "      <th>AvgChrs</th>\n",
       "      <th>AT2_04M_F0</th>\n",
       "      <th>AT2_18M_F0</th>\n",
       "      <th>AT2_04M_F10</th>\n",
       "      <th>AT2_18M_F10</th>\n",
       "      <th>AM_04M_F0</th>\n",
       "      <th>AM_18M_F0</th>\n",
       "      <th>AM_04M_F10</th>\n",
       "      <th>...</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>AM_18M_F0_P</th>\n",
       "      <th>AM_04M_F0_P</th>\n",
       "      <th>AM_18M_F10_P</th>\n",
       "      <th>AM_04M_F10_P</th>\n",
       "      <th>AT2_18M_F0_P</th>\n",
       "      <th>AT2_04M_F0_P</th>\n",
       "      <th>AT2_18M_F10_P</th>\n",
       "      <th>AT2_04M_F10_P</th>\n",
       "      <th>NormPosition</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Row.names</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>Gnai3</td>\n",
       "      <td>guanine nucleotide binding protein (G protein)...</td>\n",
       "      <td>2.675624</td>\n",
       "      <td>7.151432</td>\n",
       "      <td>7.134999</td>\n",
       "      <td>7.301802</td>\n",
       "      <td>7.126699</td>\n",
       "      <td>7.808586</td>\n",
       "      <td>7.735874</td>\n",
       "      <td>7.599455</td>\n",
       "      <td>...</td>\n",
       "      <td>38866</td>\n",
       "      <td>25.398656</td>\n",
       "      <td>25.29266</td>\n",
       "      <td>25.440647</td>\n",
       "      <td>25.149847</td>\n",
       "      <td>25.344119</td>\n",
       "      <td>25.38975</td>\n",
       "      <td>25.402401</td>\n",
       "      <td>25.485167</td>\n",
       "      <td>0.675624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>Cdc45</td>\n",
       "      <td>cell division cycle 45</td>\n",
       "      <td>15.191392</td>\n",
       "      <td>1.347537</td>\n",
       "      <td>0.520289</td>\n",
       "      <td>1.752669</td>\n",
       "      <td>1.091367</td>\n",
       "      <td>2.649199</td>\n",
       "      <td>1.417394</td>\n",
       "      <td>3.991956</td>\n",
       "      <td>...</td>\n",
       "      <td>31540</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.191392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000056</td>\n",
       "      <td>Narf</td>\n",
       "      <td>nuclear prelamin A recognition factor</td>\n",
       "      <td>10.993152</td>\n",
       "      <td>4.248040</td>\n",
       "      <td>4.163523</td>\n",
       "      <td>4.249110</td>\n",
       "      <td>4.224477</td>\n",
       "      <td>4.329426</td>\n",
       "      <td>3.781370</td>\n",
       "      <td>4.352047</td>\n",
       "      <td>...</td>\n",
       "      <td>18603</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.993152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000058</td>\n",
       "      <td>Cav2</td>\n",
       "      <td>caveolin 2</td>\n",
       "      <td>5.115437</td>\n",
       "      <td>6.162987</td>\n",
       "      <td>5.953024</td>\n",
       "      <td>6.113755</td>\n",
       "      <td>5.937759</td>\n",
       "      <td>6.605884</td>\n",
       "      <td>7.103410</td>\n",
       "      <td>6.679218</td>\n",
       "      <td>...</td>\n",
       "      <td>7930</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.115437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSMUSG00000000078</td>\n",
       "      <td>Klf6</td>\n",
       "      <td>Kruppel-like factor 6</td>\n",
       "      <td>12.048712</td>\n",
       "      <td>6.635019</td>\n",
       "      <td>6.649572</td>\n",
       "      <td>6.795728</td>\n",
       "      <td>6.626275</td>\n",
       "      <td>8.641117</td>\n",
       "      <td>8.631336</td>\n",
       "      <td>8.324152</td>\n",
       "      <td>...</td>\n",
       "      <td>8912</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.048712</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene.names  \\\n",
       "Row.names                       \n",
       "ENSMUSG00000000001      Gnai3   \n",
       "ENSMUSG00000000028      Cdc45   \n",
       "ENSMUSG00000000056       Narf   \n",
       "ENSMUSG00000000058       Cav2   \n",
       "ENSMUSG00000000078       Klf6   \n",
       "\n",
       "                                                     Gene.description  \\\n",
       "Row.names                                                               \n",
       "ENSMUSG00000000001  guanine nucleotide binding protein (G protein)...   \n",
       "ENSMUSG00000000028                             cell division cycle 45   \n",
       "ENSMUSG00000000056              nuclear prelamin A recognition factor   \n",
       "ENSMUSG00000000058                                         caveolin 2   \n",
       "ENSMUSG00000000078                              Kruppel-like factor 6   \n",
       "\n",
       "                      AvgChrs  AT2_04M_F0  AT2_18M_F0  AT2_04M_F10  \\\n",
       "Row.names                                                            \n",
       "ENSMUSG00000000001   2.675624    7.151432    7.134999     7.301802   \n",
       "ENSMUSG00000000028  15.191392    1.347537    0.520289     1.752669   \n",
       "ENSMUSG00000000056  10.993152    4.248040    4.163523     4.249110   \n",
       "ENSMUSG00000000058   5.115437    6.162987    5.953024     6.113755   \n",
       "ENSMUSG00000000078  12.048712    6.635019    6.649572     6.795728   \n",
       "\n",
       "                    AT2_18M_F10  AM_04M_F0  AM_18M_F0  AM_04M_F10  ...  \\\n",
       "Row.names                                                          ...   \n",
       "ENSMUSG00000000001     7.126699   7.808586   7.735874    7.599455  ...   \n",
       "ENSMUSG00000000028     1.091367   2.649199   1.417394    3.991956  ...   \n",
       "ENSMUSG00000000056     4.224477   4.329426   3.781370    4.352047  ...   \n",
       "ENSMUSG00000000058     5.937759   6.605884   7.103410    6.679218  ...   \n",
       "ENSMUSG00000000078     6.626275   8.641117   8.631336    8.324152  ...   \n",
       "\n",
       "                    Gene Length  AM_18M_F0_P  AM_04M_F0_P AM_18M_F10_P  \\\n",
       "Row.names                                                                \n",
       "ENSMUSG00000000001        38866    25.398656     25.29266    25.440647   \n",
       "ENSMUSG00000000028        31540     0.000000      0.00000     0.000000   \n",
       "ENSMUSG00000000056        18603     0.000000      0.00000     0.000000   \n",
       "ENSMUSG00000000058         7930     0.000000      0.00000     0.000000   \n",
       "ENSMUSG00000000078         8912     0.000000      0.00000     0.000000   \n",
       "\n",
       "                    AM_04M_F10_P  AT2_18M_F0_P  AT2_04M_F0_P  AT2_18M_F10_P  \\\n",
       "Row.names                                                                     \n",
       "ENSMUSG00000000001     25.149847     25.344119      25.38975      25.402401   \n",
       "ENSMUSG00000000028      0.000000      0.000000       0.00000       0.000000   \n",
       "ENSMUSG00000000056      0.000000      0.000000       0.00000       0.000000   \n",
       "ENSMUSG00000000058      0.000000      0.000000       0.00000       0.000000   \n",
       "ENSMUSG00000000078      0.000000      0.000000       0.00000       0.000000   \n",
       "\n",
       "                    AT2_04M_F10_P  NormPosition  \n",
       "Row.names                                        \n",
       "ENSMUSG00000000001      25.485167      0.675624  \n",
       "ENSMUSG00000000028       0.000000      0.191392  \n",
       "ENSMUSG00000000056       0.000000      0.993152  \n",
       "ENSMUSG00000000058       0.000000      0.115437  \n",
       "ENSMUSG00000000078       0.000000      0.048712  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GP_data.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
